src/assets/scripts/client/aircraft/FlightManagementSystem/Fms.js

```diff
@@ -699,11 +703,6 @@ export default class Fms {
      * @param runwayModel {RunwayModel}
      */
     replaceDepartureProcedure(routeString, runwayModel) {
-        // this is the same procedure that is already set, no need to continue
-        if (this.hasLegWithRouteString(routeString)) {
```

Comment:
> we could, potentially, attempt to replace a given flightPlan with exactly the same thing

That's what I was actually trying to do. Perhaps there is a better way. But I was saying whenever you assign a new departure runway, regenerate FROM SCRATCH the SID leg. If reassigning the same runway, or a runway where the resulting waypoints are identical, it is obviously wasted energy, but if a single fix is different (or same fix with an altitude restriction that only applies from one runway), by regenerating fresh, I can be confident we will always have the correct waypoints in place. 

Thoughts?